execution/executor-benchmark/src/block_partitioning.rs

```diff
@@ -0,0 +1,87 @@
+// Copyright © Aptos Foundation
+
+use crate::pipeline::ParToExeMsg;
+use aptos_block_partitioner::sharded_block_partitioner::ShardedBlockPartitioner;
+use aptos_crypto::HashValue;
+use aptos_logger::info;
+use aptos_types::{
+    block_executor::partitioner::{
+        CrossShardDependencies, ExecutableBlock, ExecutableTransactions,
+        TransactionWithDependencies,
+    },
+    transaction::Transaction,
+};
+use std::{
+    sync::mpsc::Receiver,
+    time::Instant,
+};
+use std::sync::mpsc::SyncSender;
+
+pub(crate) struct BlockPartitioningStage {
+    num_iterations: usize,
+    executable_block_sender: SyncSender<ParToExeMsg>,
+    maybe_exe_fin_receiver: Option<Receiver<()>>,
+    maybe_partitioner: Option<ShardedBlockPartitioner>,
+}
+
+impl BlockPartitioningStage {
+    pub fn new(
+        executable_block_sender: SyncSender<ParToExeMsg>,
+        maybe_exe_fin_receiver: Option<Receiver<()>>,
+        num_shards: usize,
+    ) -> Self {
+        let maybe_partitioner = if num_shards <= 1 {
+            None
+        } else {
+            let partitioner = ShardedBlockPartitioner::new(num_shards);
+            Some(partitioner)
+        };
+
+        Self {
+            num_iterations: 0,
```

Comment:
Is this block number? If so, let's rename it to `block_num`

Reply:
fixed